radix-engine/src/engine/track.rs

```diff
@@ -441,50 +449,68 @@ impl<'s, S: SubstateStore> Track<'s, S> {
     }
 
     /// Commits changes to the underlying ledger.
+    /// Currently none of these objects are deleted so all commits are puts
     pub fn commit(&mut self) {
-        for package_id in self.updated_packages.clone() {
-            self.ledger
-                .put_package(package_id, self.packages.get(&package_id).unwrap().clone());
+        let package_ids: Vec<PackageId> = self
+            .packages
+            .iter()
+            .map(|(address, _)| address.clone())
+            .collect();
+        for package_id in package_ids {
+            let package = self.packages.remove(&package_id).unwrap();
+            self.put_substate(&package_id, &package);
```

Comment:
Minor: There must be a better way of doing with. Otherwise, I would be disappointed with Rust. ;-)

Reply:
A quick search from me didn't show any good way :) Iteration + Removal always tricky